src/application/services/useTools.ts

```diff
@@ -60,23 +41,23 @@ export function useTools(noteTools: Ref<EditorTool[]>): {
   }
 
   watch(
-    [userEditorTools, noteTools],
+    [userEditorTools, tools],
     async () => {
-      tools.value = mergeTools(userEditorTools.value || [], noteTools.value);
+      mergedTools.value = mergeTools(userEditorTools.value || [], tools.value);
 
       /**
        * If tools are not loaded yet or empty, skip downloading their scripts
        */
-      if (tools.value.length === 0) {
+      if (mergedTools.value.length === 0) {
         return;
       }
-      toolsConnected.value = await downloadTools(tools.value);
+      noteAndUserTools.value = await editorToolsService.getTools(mergedTools.value);
     },
     { immediate: true }
   );
 
   return {
-    toolsConnected,
-    tools,
+    tools: mergedTools,
+    noteAndUserTools
```

Comment:
there are still unclear naming: what is `tools`, `mergedTools` and what is the difference between them and `noteAndUserTools`

Reply:
It was : 
```
  return {
    toolsConnected,
    tools,
  };
```
And i'm not sure what are toolsConnected and tools. What's a difference? I can return this, if they are approved names

Reply:
I think you need to refactor this service making it more straightforward